src/ripple/rpc/handlers/Unsubscribe.cpp

```diff
@@ -37,98 +37,91 @@ Json::Value doUnsubscribe (RPC::Context& context)
     InfoSub::pointer ispSub;
     Json::Value jvResult (Json::objectValue);
 
-    if (!context.infoSub && !context.params.isMember (jss::url))
+    if (! context.infoSub && ! context.params.isMember(jss::url))
     {
         // Must be a JSON-RPC call.
-        return rpcError (rpcINVALID_PARAMS);
+        return rpcError(rpcINVALID_PARAMS);
     }
 
-    if (context.params.isMember (jss::url))
+    if (context.params.isMember(jss::url))
     {
         if (context.role != Role::ADMIN)
-            return rpcError (rpcNO_PERMISSION);
+            return rpcError(rpcNO_PERMISSION);
 
-        std::string strUrl  = context.params[jss::url].asString ();
-        ispSub  = context.netOps.findRpcSub (strUrl);
-
-        if (!ispSub)
+        std::string strUrl = context.params[jss::url].asString ();
+        ispSub = context.netOps.findRpcSub (strUrl);
+        if (! ispSub)
             return jvResult;
     }
     else
     {
-        ispSub  = context.infoSub;
+        ispSub = context.infoSub;
     }
 
     if (context.params.isMember (jss::streams))
     {
+        if (! context.params[jss::streams].isArray ())
+            return rpcError (rpcINVALID_PARAMS);
+
         for (auto& it: context.params[jss::streams])
         {
-            if (it.isString ())
-            {
-                std::string streamName = it.asString ();
+            if (! it.isString())
+                RPC::inject_error(rpcSTREAM_MALFORMED, jvResult);
```

Comment:
This is injecting the error, but then going on to attempt processing of `it`, which will, of course, throw an exception when the next line calls `asString()`. I would assume you want to either `continue` or `return` after the injection.
```
            if (! it.isString())
            {
                RPC::inject_error(rpcSTREAM_MALFORMED, jvResult);
                continue;
            }
```